File: backend/Services/Models/Request/DangerModel.cs
using DAL.Entities;

namespace Services.Models.Request;

public class DangerModel : IRequestModel<Danger> {
  public DangerType Type { get; set; }
  public string Description { get; set; }
  public double Lat { get; set; }
  public double Lon { get; set; }
  public string? AddressName { get; set; }
  public string? Title { get; set; }
  public Guid CategoryId { get; set; }
  public string? OtherCategory { get; set; }
  public DateTime? ActiveAt { get; set; }

  public Danger ToEntity() {
    return new Danger {
      Type = Type,
      Description = Description,
      Lat = Lat,
      Lon = Lon,
      AddressName = AddressName,
      Title = Title,
      CategoryId = CategoryId,
      OtherCategory = OtherCategory,
      ActiveAt = ActiveAt
    };
  }
}

File: frontend/src/services/entities/DangerModel.ts
import { DangerType } from "./Danger";


export interface DangerModel {
    type?: DangerType;
    description?: string | undefined;
    lat?: number;
    lon?: number;
    addressName?: string | undefined;
    title?: string | undefined;
    categoryId?: string;
    otherCategory?: string | undefined;
    activeAt?: Date | undefined;
}
